### PR TITLE
Adding Index Settings validation before starting replication

### PR DESCRIPTION
**Pull request by gbbafna**
Signed-off-by: Gaurav Bafna <gbbafna@amazon.com>

### Description
Before starting the replication , we need to validate for correctness of index settings. This would throw validation exception before starting replication itself. 

If not done, the snapshot restore phase of replication will fail asynchronously and customer would need to stop and start the replication again without error message being communicated . 
 
### Issues Resolved

https://github.com/opensearch-project/cross-cluster-replication/issues/460
https://github.com/opensearch-project/cross-cluster-replication/issues/298
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by soosinha — DISMISSED** [review]

File: src/test/kotlin/org/opensearch/replication/integ/rest/StartReplicationIT.kt [hunk 1]
```diff
@@ -1169,6 +1169,25 @@ class StartReplicationIT: MultiClusterRestTestCase() {
...
+        createConnectionBetweenClusters(FOLLOWER, LEADER)
+
+        val createIndexResponse = leaderClient.indices().create(CreateIndexRequest(leaderIndexName), RequestOptions.DEFAULT)
+        assertThat(createIndexResponse.isAcknowledged).isTrue()
+        val settings = Settings.builder()
+            .put("index.data_path", "/random-path/invalid-setting")
+            .build()
+
+        assertThatThrownBy {
+            followerClient.startReplication(StartReplicationRequest("source", leaderIndexName, followerIndexName, settings = settings))
```
Comment (saikaranam-amazon) [inline src/test/kotlin/org/opensearch/replication/integ/rest/StartReplicationIT.kt:1185]: Can we validated status code as well?

File: src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt [hunk 2]
```diff
@@ -31,6 +31,27 @@ object ValidationUtil {
...
     private val log = LogManager.getLogger(ValidationUtil::class.java)
 
+    fun validateIndexSettings(
+        environment: Environment,
+        followerIndex: String,
+        leaderSettings: Settings,
+        overriddenSettings: Settings,
+        metadataCreateIndexService: MetadataCreateIndexService
+    ) {
+        val settingsList = arrayOf(leaderSettings, overriddenSettings)
```
Comment (saikaranam-amazon) [inline src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt:41]: Please validate to see if the behavior is as expected, due to open bug on replication side: https://github.com/opensearch-project/cross-cluster-replication/issues/298?

File: src/test/kotlin/org/opensearch/replication/integ/rest/StartReplicationIT.kt (hunk 1 above)
Comment (gbbafna) [reply src/test/kotlin/org/opensearch/replication/integ/rest/StartReplicationIT.kt:1185]: done .

File: src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt (hunk 2 above)
Comment (gbbafna) [reply src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt:41]: I have fixed this bug in this PR as well. Now we are retrieving the default settings as well .

**Review by saikaranam-amazon — APPROVED** [review]

**Review by soosinha — APPROVED** [review]

**Pull request merged**